packages/compiler-cli/ngcc/src/analysis/module_with_providers_analyzer.ts

```diff
@@ -108,14 +207,13 @@ export class ModuleWithProvidersAnalyzer {
       throw new Error(`No typings declaration can be found for the referenced NgModule class in ${
           fn.declaration.getText()}.`);
     }
-    if (!ts.isClassDeclaration(dtsNgModule) || !hasNameIdentifier(dtsNgModule)) {
+    if (!isNamedClassDeclaration(dtsNgModule)) {
       throw new Error(`The referenced NgModule in ${
           fn.declaration
               .getText()} is not a named class declaration in the typings program; instead we get ${
           dtsNgModule.getText()}`);
     }
-
-    return {node: dtsNgModule, known: null, viaModule: null, identity: null};
+    return new Reference(dtsNgModule, ngModule.bestGuessOwningModule);
```

Comment:
```suggestion
    return new Reference(dtsNgModule, /* owningModule */ null);
```

This is always going to be `null` I believe, because of the early return above.